### ISSUE TITLE
Access error when installing packages

### ISSUE THREAD

**Issue by rafaelrinaldi** [npm/npm#10683]
For some weird reason I'm getting access error for any packages that I try to install.

For example, if I try to install `stylus` on a local project:

```sh
❯ npm install stylus --save-dev
npm WARN checkPermissions Missing write access to /Users/rafaelrinaldi/work/project/node_modules
/Users/rafaelrinaldi/work/project
└── stylus@0.52.4

npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/Cellar/node/5.1.0/bin/node" "/usr/local/bin/npm" "install" "stylus" "--save-dev"
npm ERR! node v5.1.0
npm ERR! npm  v3.3.12
npm ERR! path /Users/rafaelrinaldi/work/project/node_modules
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access

npm ERR! Error: EACCES: permission denied, access '/Users/rafaelrinaldi/work/project/node_modules'
npm ERR!     at Error (native)
npm ERR!  { [Error: EACCES: permission denied, access '/Users/rafaelrinaldi/work/project/node_modules']
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/Users/rafaelrinaldi/work/project/node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/rafaelrinaldi/work/project/npm-debug.log
```

Using `sudo` should probably fix this but it makes no sense to require admin privileges to install a package on a local project. Any idea of what's happening?

I've installed `node` via Homebrew (which installs `npm` automatically), and I'm on a OS X El Capitain.

Here's the log file:

```sh
520 silly generateActionsToTake Starting
521 silly install generateActionsToTake
522 warn checkPermissions Missing write access to /Users/rafaelrinaldi/work/project/node_modules
523 silly rollbackFailedOptional Starting
524 silly rollbackFailedOptional Finishing
525 silly runTopLevelLifecycles Starting
526 silly runTopLevelLifecycles Finishing
527 silly install printInstalled
528 verbose stack Error: EACCES: permission denied, access '/Users/rafaelrinaldi/work/project/node_modules'
528 verbose stack     at Error (native)
529 verbose cwd /Users/rafaelrinaldi/work/project
530 error Darwin 15.0.0
531 error argv "/usr/local/Cellar/node/5.1.0/bin/node" "/usr/local/bin/npm" "install" "stylus" "--save-dev"
532 error node v5.1.0
533 error npm  v3.3.12
534 error path /Users/rafaelrinaldi/work/project/node_modules
535 error code EACCES
536 error errno -13
537 error syscall access
538 error Error: EACCES: permission denied, access '/Users/rafaelrinaldi/work/project/node_modules'
538 error     at Error (native)
538 error  { [Error: EACCES: permission denied, access '/Users/rafaelrinaldi/work/project/node_modules']
538 error   errno: -13,
538 error   code: 'EACCES',
538 error   syscall: 'access',
538 error   path: '/Users/rafaelrinaldi/work/project/node_modules' }
539 error Please try running this command again as root/Administrator.
540 verbose exit [ -13, true ]
```

**Comment by rafaelrinaldi** [comment]
Well, I have removed the `node_modules` folder, tried again and it worked. Closing the issue.

**Comment by macOneOone** [comment]
you can also use sudo before npm

**Comment by taylorstine** [comment]
Nah don't do `sudo npm install -g`, it could be dangerous.  Do this instead https://docs.npmjs.com/getting-started/fixing-npm-permissions

**Comment by macOneOone** [comment]
ok, you're right

**Comment by valoricDe** [comment]
I found this error via google because I had the same problem on Docker for Windows. Appearantly the file index was corrupted. Restarting the machine helped.

**Comment by codingwesley** [comment]
@rafaelrinaldi  thanks your answer!

**Comment by paulresdat** [comment]
I recently had this issue on my Mac using High Sierra where I had a strange write error having an existing `@angular` directory with no `cli` directory in it.  Most likely because I didn't install it globally in the manner I'm supposed to AND I shouldn't use `sudo` or shouldn't have to I should say.  I was able to fix the issue this way:

`cd /usr/local/lib` where the global node_module folder is found

`sudo chown -R username:group node_modules` to ensure it was accessible by my user without administrative privileges.  I used `[myname]:admin` but I would double check by getting the username group from `ls -la ~` and make sure.

Then I `cd node_modules; rm -rf \@angular` to remove the module from the folder

Then I finally ran `npm install -g @angular/cli` and it then worked for me fine.
